### DISCUSSION TITLE
Have Authorizers been executed even if no Authenticator is enabled

### DISCUSSION THREAD

**Discussion by dev-indb** [SoftInstigate/restheart#417] [Ideas]
I wrote a custom `Authorizer` plugin for RESTHeart **5.1.5** version but method `isAllowed` is never called when I try to work directly in my IDE and [RESTHeart project source code](https://github.com/SoftInstigate/restheart/archive/refs/tags/5.1.5.zip)
I can see the plugin is correctly registered in log file.

`fileAclAuthorizer` is enabled in the conf file and the attribute `enabledByDefault` is equals to `true` of `@registerPlugin` annotation

![image](https://user-images.githubusercontent.com/23722884/137790526-6aa7d67f-884c-4c2b-ab7d-40c4e32c2b27.png)

I found a long entry with this message : "No Authentication Mechanisms defined"

![image](https://user-images.githubusercontent.com/23722884/137790659-6c122a57-3fcc-4899-8d44-cb6ffcaaa2ea.png)

I enabled tokenBasicAuthMechanism and the plugin works correctly.

![image](https://user-images.githubusercontent.com/23722884/137790806-9a8bd2fb-7fa5-46d3-bfb9-49453cb12093.png)

In my plugin, I return `false` to `isAuthenticationRequired`
That's why the plugin code is not processed by the handler

![image](https://user-images.githubusercontent.com/23722884/137790871-fb75ceeb-14f1-47eb-bf41-1d2eedab262b.png)

This means that the authorizer code is processed if and only if there is at least one authentication mechanism.
That sounds good to me

Maybe an `isAuthorizationRequired` method could load the security context too ?

![image](https://user-images.githubusercontent.com/23722884/137791246-0cf6b419-8a28-437c-8026-0d08ba8023a3.png)

**Comment by ujibang** [comment]
Currently the authorization logic is executed if and only if there is at least one authentication mechanism.

We will modify this behavior in RESTHeart 6.2 to execute authorization logic if the request handling service is secured (i.e. its definition specifies the attribute `secure=true` or its configuration specifies `secured: true`)

This will allow to configure no auth mechanism but still authorizes requests (for instance, based on remote ip address).

To reproduce the current behavior the authorizer `fullAuthorizer` (that just authorizes any request) can be enabled.

**Comment by ujibang** [comment]
Done in commit 23bd40853f865941361f9d3712a4883f6940c7cc, will be part of RESTHeart 6.2

@dev-indb You can try it now using [snapshot builds](https://github.com/softInstigate/restheart#snapshot-builds)

**Reply by dev-indb** [reply]
Thank you for all this work @ujibang
I'll test it as soon as possible and give you feedback
